awscli/customizations/configure/set.py

```diff
@@ -60,43 +64,44 @@ def _run_main(self, args, parsed_globals):
             # profile (or leave it as the 'default' section if
             # no profile is set).
             if self._session.profile is not None:
-                section = profile_to_section(self._session.profile)
+                profile = self._session.profile
         else:
             # First figure out if it's been scoped to a profile.
             parts = varname.split('.')
             if parts[0] in ('default', 'profile'):
                 # Then we know we're scoped to a profile.
                 if parts[0] == 'default':
-                    section = 'default'
+                    profile = 'default'
                     remaining = parts[1:]
                 else:
                     # [profile, profile_name, ...]
-                    section = profile_to_section(parts[1])
+                    profile = parts[1]
                     remaining = parts[2:]
                 varname = remaining[0]
                 if len(remaining) == 2:
                     value = {remaining[1]: value}
             elif parts[0] not in PREDEFINED_SECTION_NAMES:
                 if self._session.profile is not None:
-                    section = profile_to_section(self._session.profile)
+                    profile = self._session.profile
                 else:
                     profile_name = self._session.get_config_variable('profile')
                     if profile_name is not None:
-                        section = profile_name
+                        profile = profile_name
                 varname = parts[0]
                 if len(parts) == 2:
                     value = {parts[1]: value}
             elif len(parts) == 2:
                 # Otherwise it's something like "set preview.service true"
                 # of something in the [plugin] section.
-                section, varname = parts
-        config_filename = os.path.expanduser(
-            self._session.get_config_variable('config_file'))
-        updated_config = {'__section__': section, varname: value}
+                profile, varname = parts
```

Comment:
Talked about this offline but it's a little weird to be calling the predefined sections profiles. But sounds like it will be tricky to switch the names here so let's leave a comment for future reference.